### ISSUE TITLE
[chore] Separate website/elements/apps from library?

### ISSUE THREAD

**Issue by LeaVerou** [color-js/color.js#434]
The current structure has served us well but we're at the point where we need to separate the two.

In terms of tests, presumably the JS tests will stay with the library, while the HTML tests (old and new) will live with the website.

Both will still deploy to the web, I’m thinking `dev.colorjs.io` for the library repo.

**Comment by LeaVerou** [comment]
Where would the API docs live? Presumably they need access to the source, but perhaps that can be done by installing `color.js` as an npm package on the website repo and pointing typedoc to that?

**Comment by MysteryBlokHed** [comment]
> Where would the API docs live? Presumably they need access to the source, but perhaps that can be done by installing `color.js` as an npm package on the website repo and pointing typedoc to that?

That should work if we want to keep all website stuff together. Alternatively there could also be a Git submodule that links back to this repo, but that might be overcomplicating things.

**Comment by LeaVerou** [comment]
Yeah, my opinion on this is that submodules are plague and must be avoided at all costs 😛

**Comment by LeaVerou** [comment]
I’ve been thinking about this and I think it makes sense to also separate apps and elements into separate repos / separate npm packages / separate subdomains.

Especially as I plan a bit of a revamp of these, what do you think about separating them?
We could create an npm namespace and publish them as `@colorjs/apps` and `@colorjs/elements`. 
Later down the line we may even want to separate elements into separate packages, but let’s cross that bridge when we get to it.

**Comment by LeaVerou** [comment]
As an update, I have split elements and apps into their own repos and websites (https://apps.colorjs.io and https://elements.colorjs.io respectively). The website will be harder to split out though…

**Comment by LeaVerou** [comment]
Ok, thinking about this some more, I have a plan:
- We split into two repos: `core` (which we can deploy, e.g. to `dev.colorjs.io`) and `colorjs.io` (names TBB, suggestions welcome)
- `core` will contain `test`, `tests`, `src`, `types`, `scripts`, `CONTRIBUTING.md`, `.editorconfig`, ESLint config, tsconfig, LICENSE, README.md
- `colorjs.io` will contain: Notebook, 11ty files, assets, `get/`
- Unsure: `benchmarks`, Typedoc, 

Thoughts?

**Comment by MysteryBlokHed** [comment]
I think it'd be ideal to have tests/benchmarks alongside the source code, but I'm not sure how practical it'd be to host them on the website if we do that. If there's an easy way to set up the website to use content from multiple repositories (besides Git submodules which we've decided against), then I'd opt for putting those directories with the source code. Otherwise I'd say they should go with the rest of the website content.

**Comment by LeaVerou** [comment]
Depends on what you mean by "the website". We can easily deploy the code repo to a separate subdomain (e.g. `dev.colorjs.io`)

**Comment by MysteryBlokHed** [comment]
That'd be the best way to do it then imo

**Comment by justinfagnani** [comment]
fwiw, I find an npm monorepo to be a really nice way to handle multiple packages like this. You get atomic commits of changes across packages, a pretty easy checkout-to-dev workflow, centralized build/lint/test scripts, etc.
